### ISSUE TITLE
Using children operator in a query model definition

### ISSUE THREAD

**Issue by caristu** [OSBI/saiku#605]
Hello,

  using Saiku version 2.6, we were able to define a Selection in a query model with members using the CHILDREN operator. For example:

```
<?xml version="1.0" encoding="UTF-8"?>
<Query name="99B20355C9F047DD8C970EF0A12AF3BCBBB92B8E043249AFA5187CF0DBA90B81" type="QM" connection="Openbravo" cube="55D5D82E42764005A65EAFC1BE33B114" catalog="Openbravo" schema="Openbravo">
    <QueryModel>
        <Axes>
            <Axis location="ROWS" nonEmpty="true">
                <Dimensions>
                    <Dimension name="account" hierarchizeMode="PRE" hierarchyConsistent="true">
                        <Inclusions>
                            <Selection dimension="account" type="member" node="[account].[FF30CF29CE614360AF85020438BFE328].[D883C3FFB25F46D88F1E35F08204F143]" operator="CHILDREN" />
                            <Selection dimension="account" type="member" node="[account].[FF30CF29CE614360AF85020438BFE328]" operator="CHILDREN" />
                        </Inclusions>
                        <Exclusions />
                    </Dimension>
                </Dimensions>
            </Axis>
            <Axis location="COLUMNS" nonEmpty="true">
                <Dimensions>
                    <Dimension name="accountingDate">
                        <Inclusions>
                            <Selection dimension="accountingDate" type="member" node="[accountingDate].[2015]" operator="MEMBER" />
                        </Inclusions>
                        <Exclusions />
                    </Dimension>
                </Dimensions>
            </Axis>
            <Axis location="FILTER" nonEmpty="false">
                <Dimensions>
                    <Dimension name="Measures">
                        <Inclusions>
                            <Selection dimension="Measures" type="member" node="[Measures].[balance]" operator="MEMBER" />
                        </Inclusions>
                        <Exclusions />
                    </Dimension>
                </Dimensions>
            </Axis>
        </Axes>
    </QueryModel>
    <MDX />
    <Properties>
        <Property name="saiku.ui.render.mode" value="table" />
        <Property name="org.saiku.query.explain" value="true" />
        <Property name="saiku.olap.query.nonempty.columns" value="true" />
        <Property name="saiku.olap.query.nonempty.rows" value="true" />
        <Property name="org.saiku.connection.scenario" value="false" />
        <Property name="saiku.ui.formatter" value="flat" />
        <Property name="saiku.olap.query.automatic_execution" value="true" />
        <Property name="saiku.olap.query.drillthrough" value="true" />
        <Property name="saiku.olap.query.filter" value="true" />
        <Property name="saiku.olap.query.limit" value="true" />
        <Property name="saiku.olap.query.nonempty" value="true" />
    </Properties>
</Query>
```

  This allowed us to define different queries specifying which nodes of a hierarchy we wanted to display.

  Now we are using Saiku version 3.7.4, and we can not use a query like the one defined above because the QueryConverter throws the following exception: 

    _org.saiku.olap.util.exception.SaikuIncompatibleException: Cannot convert member selection using operator: CHILDREN
	at org.saiku.olap.util.QueryConverter.convertDimension(QueryConverter.java:91)
	at org.saiku.olap.util.QueryConverter.convertAxis(QueryConverter.java:41)
	at org.saiku.olap.util.QueryConverter.convertQuery(QueryConverter.java:31)
	at org.saiku.service.olap.ThinQueryService.convertQuery(ThinQueryService.java:719)_

  -   Why this kind of query definition is incompatible? There exists any workaround in the XML query definition so we can continue displaying the hierarchies as before?
  -   Is it possible to define the same query but based on the current JSON query model definition?

**Comment by caristu** [comment]
Hello,

  as previously described, it looks like in version 3.7.4 the QueryConverter[1] class is precisely avoiding to build an XML query model defining a member with an operator different from MEMBER. This seemed to be supported in version 2.6.

  We have also tried to generate a query model definition using the CHILDREN operator in a member but based on the new JSON definition. But it seems that is neither possible.

  After more research, we have also found that the "Show Children" option is not available in the Saiku User Interface. According to the following link[2] it was not migrated to version 3.7.4

  This option was available in version 2.6. Could be this related with the reported problem? Is it supported to define a member with the CHILDREN operator in a query model definition?

[1] https://github.com/OSBI/saiku/blob/18088504272927a9d016e6550a32961db4f6526a/saiku-core/saiku-service/src/main/java/org/saiku/olap/util/QueryConverter.java#L91 
[2] https://groups.google.com/a/saiku.meteorite.bi/forum/#!topic/user/2B1QDOW3_LQ

**Comment by buggtb** [comment]
It got disabled in the replacement query model we built for olap4j. I can't remember the reason precisely but it was to do with member navigation and ordering IIRC. You can file a patch, but I don't see it reappearing any time soon otherwise.
